### PR TITLE
Automatic tagging support

### PR DESCRIPTION
**Pull request by JanLukasGithub**
Automatically tags new/edited proposals based on the location of the contained stations

### CODE REVIEW

**Comment by JanLukasGithub** [comment]
> Thanks! Great that we finally have automatic place-related tagging.
> 
> Two questions:
> 
>     1. Shouldn't we always use `let` keyword instead of `var` in modern JS when defining variables?

Yes, I will change that.

> 
>     2. The changes in _mtl-proposal-form.php_ still contain local links. This should be changed to correct relative links, if I'm not missing something.

Can you comment on the exact code line where that is? From what I see, only the links to the data source aren't relative, which is desired to allow for greater flexibility. With this you can just upload the files to wordpress media, and paste their link into the backend setting. But I could change this to only allow stuff in the [wp-content](https://developer.wordpress.org/reference/functions/content_url/) directory?

File: modules/mtl-proposal-form/mtl-proposal-form.php [hunk 1]
```diff
@@ -265,6 +265,11 @@ function mtl_proposal_form_output( $atts ){
 			}
 			$output .= '}; </script>'."\r\n";
 			$output .= '<script type="text/javascript"> '.$output_later.' var centerLon = "'.$mtl_options['mtl-center-lon'].'"; var centerLat = "'.$mtl_options['mtl-center-lat'].'"; var standardZoom = '.$mtl_options['mtl-standard-zoom'].'; </script>'."\r\n";
+
+			// TODO make this a setting
+			$output .= '<script type="text/javascript"> var countrySource = \''.str_replace(array("\r", "\n"), "", file_get_contents('http://localhost/wordpress/wp-content/uploads/VG5000_LAN.geojson')).'\';'."\r\n";
```
Comment (winnewoerp) [inline modules/mtl-proposal-form/mtl-proposal-form.php]: Why are there static links here? If the GeoJSON boundary files are to be used in static form within the theme, they should be included to the theme directly, which is completely fine as a temporary solution. But IMHO there should be no static links to anywhere above theme folder level.

File: modules/mtl-proposal-form/mtl-proposal-form.php [hunk 2]
```diff
@@ -263,10 +263,9 @@ function mtl_proposal_form_output( $atts ){
 			$output .= '}; </script>'."\r\n";
 			$output .= '<script type="text/javascript"> '.$output_later.' var centerLon = "'.$mtl_options['mtl-center-lon'].'"; var centerLat = "'.$mtl_options['mtl-center-lat'].'"; var standardZoom = '.$mtl_options['mtl-standard-zoom'].'; </script>'."\r\n";
 
-			// TODO make this a setting
-			$output .= '<script type="text/javascript"> var countrySource = \''.str_replace(array("\r", "\n"), "", file_get_contents('http://localhost/wordpress/wp-content/uploads/VG5000_LAN.geojson')).'\';'."\r\n";
-			$output .= 'var stateSource = \''.str_replace(array("\r", "\n"), "", file_get_contents('http://localhost/wordpress/wp-content/uploads/VG5000_STA.geojson')).'\';'."\r\n";
-			$output .= 'var districtSource = \''.str_replace(array("\r", "\n"), "", file_get_contents('http://localhost/wordpress/wp-content/uploads/VG5000_KRS.geojson')).'\'; </script>'."\r\n";
+			$output .= '<script type="text/javascript"> var countrySource = \''.str_replace(array("\r", "\n"), "", file_get_contents($mtl_options3['mtl-country-source'])).'\';'."\r\n";
```
Comment (winnewoerp) [inline modules/mtl-proposal-form/mtl-proposal-form.php:266]: Ah sorry, here we go. I did not see that it wasn't the last commit of the PR. Sorry.

**Comment by winnewoerp** [comment]
> > Thanks! Great that we finally have automatic place-related tagging.
> > Two questions:
> > ```
> > 1. Shouldn't we always use `let` keyword instead of `var` in modern JS when defining variables?
> > ```
> 
> Yes, I will change that.
> 
> > ```
> > 2. The changes in _mtl-proposal-form.php_ still contain local links. This should be changed to correct relative links, if I'm not missing something.
> > ```
> 
> Can you comment on the exact code line where that is? From what I see, only the links to the data source aren't relative, which is desired to allow for greater flexibility. With this you can just upload the files to wordpress media, and paste their link into the backend setting. But I could change this to only allow stuff in the [wp-content](https://developer.wordpress.org/reference/functions/content_url/) directory?

Solved, see my previous comments.

**Comment by JanLukasGithub** [comment]
`var` declarations have now been changed to `let` declarations. LGTM

**Review by winnewoerp — APPROVED** [review]

**Comment by winnewoerp** [comment]
Thank you, LGTM as well!

**Pull request merged**